tests/misc/tests/misc.js

```diff
@@ -1632,4 +1632,576 @@ describe("misc", () => {
+          },
+        });
+        assert.ok(false);
+      } catch (err) {
+        assert.equal(err.code, 2018);
+      }
+    });
+
+    it("Mint Constraint Test(no init) - throws if mint::mint_authority mismatch", async () => {
+      const mint = anchor.web3.Keypair.generate();
+      await program.rpc.testInitMint({
+        accounts: {
+          mint: mint.publicKey,
+          payer: program.provider.wallet.publicKey,
+          systemProgram: anchor.web3.SystemProgram.programId,
+          tokenProgram: TOKEN_PROGRAM_ID,
+          rent: anchor.web3.SYSVAR_RENT_PUBKEY,
+        },
+        signers: [mint],
+      });
+
+      // mintAuthority mismatch
+      const fakeAuthority = Keypair.generate();
+      try {
+        await program.rpc.testMintConstraint(6, {
+          accounts: {
+            mint: mint.publicKey,
+            mintAuthority: fakeAuthority.publicKey,
+            freezeAuthority: program.provider.wallet.publicKey,
+            payer: program.provider.wallet.publicKey,
+            systemProgram: anchor.web3.SystemProgram.programId,
+            tokenProgram: TOKEN_PROGRAM_ID,
+            rent: anchor.web3.SYSVAR_RENT_PUBKEY,
+          },
+        });
+        assert.ok(false);
+      } catch (err) {
+        assert.equal(err.code, 2016);
+      }
+    });
+
+    it("Mint Constraint Test(no init) - throws if mint::freeze_authority mismatch", async () => {
+      const mint = anchor.web3.Keypair.generate();
+      await program.rpc.testInitMint({
+        accounts: {
+          mint: mint.publicKey,
+          payer: program.provider.wallet.publicKey,
+          systemProgram: anchor.web3.SystemProgram.programId,
+          tokenProgram: TOKEN_PROGRAM_ID,
+          rent: anchor.web3.SYSVAR_RENT_PUBKEY,
+        },
+        signers: [mint],
+      });
+
+      const fakeAuthority = Keypair.generate();
+      // freezeAuthority mismatch
+      try {
+        await program.rpc.testMintConstraint(6, {
+          accounts: {
+            mint: mint.publicKey,
+            mintAuthority: program.provider.wallet.publicKey,
+            freezeAuthority: fakeAuthority.publicKey,
+            payer: program.provider.wallet.publicKey,
+            systemProgram: anchor.web3.SystemProgram.programId,
+            tokenProgram: TOKEN_PROGRAM_ID,
+            rent: anchor.web3.SYSVAR_RENT_PUBKEY,
+          },
+        });
+        assert.ok(false);
+      } catch (err) {
+        assert.equal(err.code, 2017);
+      }
+    });
+
+    it("Mint Constraint Test(no init) - can write only mint::decimals", async () => {
+      const mint = anchor.web3.Keypair.generate();
+      await program.rpc.testInitMint({
+        accounts: {
+          mint: mint.publicKey,
+          payer: program.provider.wallet.publicKey,
+          systemProgram: anchor.web3.SystemProgram.programId,
+          tokenProgram: TOKEN_PROGRAM_ID,
+          rent: anchor.web3.SYSVAR_RENT_PUBKEY,
+        },
+        signers: [mint],
+      });
+
+      // right
```

Comment:
no need for this here. test title is sufficient

Reply:
Okay